dandi/metadata.py

```diff
@@ -319,7 +308,17 @@ def nwb2asset(nwb_path, digest=None, digest_type=None):
 
 
 def metadata2asset(metadata):
-    return extract_model(models.BareAssetMeta, metadata)
+    m = extract_model(models.BareAssetMeta, metadata)
+    derived_from = None
+    for field in ("subject_id", "tissue_sample_id", "slice_id", "cell_id"):
+        if metadata.get(field) is not None:
+            derived_from = [
+                models.BioSample(
```

Comment:
Here identifier wouldn't provide semantic of what that identifier of. 
@satra - should we need to define classes for each one of those types of biosample at different levels ?  For subject_id we already have `Participant` in the model (but it is not a BioSample, so we would need to either make it a subclass of  BioSample or may be better allow wasDerivedFrom to be
```python
    wasDerivedFrom: Optional[Union(BioSample, Participant)]] = Field(None, nskey="prov")
```
instead of current
```python
    wasDerivedFrom: Optional[List["BioSample"]] = Field(None, nskey="prov")
```
(unless you see the need for list with the hierarchy), and then define subclasses of `BioSample`: `Tissue`, `Slice`, `Cell`.  WDYT?

Reply:
I see that we have already 
```
    wasAttributedTo: List[Participant] = Field(
        None, description="Participant(s) to which this file belongs to", nskey="prov"
    )
```
so I guess we should not place subject id in `wasDerivedFrom` and rely on participant be linked only in `wasAttributedTo`, correct @satra?

Reply:
correct.  for BioSample, we can either create subclasses or set `sampleType` - it's a new field i added this morning.

Reply:
Assuming we want to set the `sampleType` on each `BioSample` in `wasDerivedFrom`, what would the various fields of the `SampleType` be?

Reply:
I was wondering about that (`sampleType`) myself but forgot to ask @satra . as far as I see it boils down to use of an ontology, which we do not really do yet, so I would not sure if we can do anything about it ATM.  But it would be nice for all those referring to OBI to at least have a link to it alongside somewhere (I failed to find the ultimate source ATM via google).